CodeU_Assignement_1/src/codeu_assignement_1/main.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_1;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+
+/**
+ *
+ * @author Karine
+ */
+public class main {
+    /**
+     * @param args the command line arguments
+     * 
+     * Gives a basic interface to do some partial testing to the code
+     * 
+     */
+    public static void main(String[] args) {
+        if (args.length == 0) // run tests!
+        {
+            run_tests_Q1();
+            run_tests_Q2();
+            return;
+        }
+        
+        // TODO code application logic here
+        System.out.println("Parsing Input... (String1, String2, k2Last and a list of elements)");      
+        if (args.length!=4)
+        {
+            System.out.println("Error: " + args.length 
+                    + " parameters only. Expect somthing like: Listen Silen 5 1,2,3,4,5,6,7");      
+            return;
+        }
+         
+        // Q1: Run and result
+        System.out.println("Q1: " + args[0] 
+                + (q1_permutation(args[0],args[1]) ? " is " : " isn't ") 
+                + "a permutation of " + args[1]);
+        
+        // Q2: Run and result
+        int k2last = 0;
+        try {
+            k2last = Integer.parseInt(args[2]);
+        } catch (NumberFormatException e) {
+            System.out.println("Error: " + args[2] 
+                    + " is not a number, shall be the k-th to the last location");      
+            return;
+        }
+        
+        // Create a list:
+        String[] list_user = args[3].split(",");
+        SinglyLinkedList<String> intput_list;
+        intput_list = new SinglyLinkedList<>(Arrays.asList(list_user));
+        
+        // Q2: Run and result
+        String res = q2_getKthElement(intput_list, k2last); // Incase of error will throw exception before print
+        System.out.println("Q2: the " + k2last + "-th to the last element is " + res);
+    }
+    
+    /*
+    * Q1: doesn't need a class, a simple method
+    * Input is taken from the main parameters
+    * Output to the screen
+    */
+    static boolean q1_permutation(String s1, String s2) 
+    {
+        // Sanity checks
+        if (s1 == null || s2 == null)
+            if (s1 == null && s2 == null)
+                return true;
+            else 
+                return false;
+        
+        if (s1.length() != s2.length()) return false;
+        if (s1.equals(s2)) return true;
+                
+        /* Start the algorithm : */
+        
+        // To array of chars
+        char[] t1 = s1.toLowerCase().toCharArray();
+        char[] t2 = s2.toLowerCase().toCharArray();
+        
+        // Order
+        Arrays.sort(t1);
+        Arrays.sort(t2);
+        
+        // Compare
+        return Arrays.equals(t2, t1);
+    } // End Q1

```

Comment:
Your suggested algorithm can also be adapted to work with the full 16-bit range of Character. You can take a look at your colleagues implementations to see how some of them have done O(n) solutions for the full range.